warp10/src/main/java/io/warp10/script/functions/MACROMAPPER.java

```diff
@@ -138,37 +139,52 @@ public Object apply(Object[] args) throws WarpScriptException {
       //
       
       stack.exec(this.macro);
-      
-      //
+
+      return TransformMacroMapperOutput();
+    }
+
+    private Object TransformMacroMapperOutput() throws WarpScriptException {
```

Comment:
use camelCase
rather format or collect than transform
stack.exec can also be included in this method (and maybe stack.push)